src/Native/Unix/CMakeLists.txt

```diff
@@ -90,7 +90,9 @@ if (UPPERCASE_CMAKE_BUILD_TYPE STREQUAL DEBUG)
       set(CMAKE_SHARED_LINKER_FLAGS_DEBUG "${CMAKE_SHARED_LINKER_FLAGS_DEBUG} ${CLR_SANITIZE_LINK_FLAGS} -Wl,--gc-sections")
     endif ()
 elseif (UPPERCASE_CMAKE_BUILD_TYPE STREQUAL RELEASE)
-    if (CMAKE_SYSTEM_PROCESSOR STREQUAL armv7l AND DEFINED ENV{CROSSCOMPILE})
+    # Use O1 option when the clang version is smaller than 3.9
+    # Otherwise use O3 option in release build
+    if (CMAKE_SYSTEM_PROCESSOR STREQUAL armv7l AND DEFINED ENV{CROSSCOMPILE} AND CLANG_VERSION LESS 3.9)
```

Comment:
@janvorli Okay I will try to use the CMAKE_CXX_COMPILER_VERSION.
@gkhanna79 Because we still support to build corefx with clang 3.6 or clang 3.5 as before. Although the CI machine will use clang3.9, other developers may not use the clang 3.9. I think we need to consider developer using the same version(clang3.6 or clang 3.5) as we do not fix the clang version to 3.9 or higher so user can choose.